bugsnag/tornado/__init__.py

```diff
@@ -1,13 +1,43 @@
 import tornado
 from tornado.web import RequestHandler
 from tornado.web import HTTPError
-
+import urllib.parse
 import bugsnag
+import json
 
 
 class BugsnagRequestHandler(RequestHandler):
-    def _handle_request_exception(self, exc):
+    def add_tornado_request_to_notification(self, notification):
+        if not hasattr(self, "request"):
+            return
+
+        try:
+            request_tab = {
+                'method': self.request.method,
+                'path': self.request.uri,
+                'GET': {},
+                'POST': {},
+                'url': '{0}://{1}{2}'.format(self.request.protocol,
+                                             self.request.host,
+                                             self.request.uri),
+            }
+            if (len(self.request.body) > 0):
+                headers = self.request.headers
+                body = self.request.body.decode('utf-8')
+                aj = 'application/json'
+                if headers.get('Content-Type').lower().startswith(aj):
+                    if request_tab["method"] == "POST":
+                        request_tab["POST"] = json.loads(body)
+                else:
+                    request_tab["POST"] = urllib.parse.parse_qs(body)
 
+            notification.add_tab("request", request_tab)
+        except Exception:
+            pass
+
+        notification.add_tab("environment", dict(self.request.headers))
```

Comment:
I think this is intended to be the WSGI environment, rather than just the request headers. Something like this should work:

```suggestion
        notification.add_tab("environment", tornado.wsgi.WSGIContainer.environ(self.request))
```